WordPress/gutenberg
# Update sidebar title + icon + site title alignment

I'm unsure when this visual regression occurred, but the sidebar title has slipped out of alignment with the icons it often appears next to, and with the site title. It appears to be resting on a different plain to the icons, which is creating some tension between the `<` button and the title.

This PR addresses these details, illustrated below:


<img width="1214" alt="Screenshot 2024-05-31 at 17 01 01" src="https://github.com/WordPress/gutenberg/assets/846565/782d0cdb-e67d-42cb-874b-ce9137990037">


### Testing Instructions
1. Open the Site Editor
2. Check the sidebar title and icons are centrally aligned along the X axis.
3. Check the sidebar title and site title are left aligned on the Y axis.
